### ISSUE TITLE
Invalid typescript type when creating dns record of type NS

### ISSUE THREAD

**Issue by bxgrant** [cloudflare/cloudflare-typescript#159]
I am using 3.0.0-beta.3

I am trying to create a NS dns record.  

When calling client.dns.records.create we are meant to pass in an object of type `RecordCreateParams`.  According to the underlying API, this should be able to take a `content` attribute and a `type` attribute.  Content will be the NS to point to.  Type should be set to the value `NS`.

However, `RecordCreateParams` doesn't include a `content` attribute in the root (it's there buried in an attribute named `data` and if that's what I'm supposed to use then fine, but it doesn't seem like that's correct according to the underlying API or the code if I'm reading it right).

Also, `RecordCreateParams` has the `type` attribute hardcoded to `URI`.  Am I thinking about this wrong?  This seems like a bug in your type.

Here's my code and what I had to do.  Note I have not yet run this code as I feared jacking things up not knowing if this cloudflare library is really a good idea or not or whether I need to fall back to just hitting the underlying APIs directly myself.  Thanks for the attention.

```typescript
await client.dns.records.create({
    zone_id: zoneId,
    name: fullDomanName,

    // The type for RecordCreateParams says this has to be URI but the API itself accepts NS
    type: 'NS',

    // The type for RecordCreateParams doesn't allow this but the API does
    content: nsName
} as unknown as RecordCreateParams);
```

**Comment by jacobbednarz** [comment]
thanks for trying out the beta!

looking into this, it appears the `content` value is missing on the [`dns.RecordCreateParams`](https://github.com/cloudflare/cloudflare-typescript/blob/192d30273a788520fedc4d3b994d20a3f0d319ad/src/resources/dns/records.ts#L14699-L14755) type however, it is definitely present on the [`dns.NS` interface](https://github.com/cloudflare/cloudflare-typescript/blob/573e0d23589bcc9e5f4cbbc1a960ab16c8eb26f0/src/resources/dns/records.ts#L1508).

confirming it all aligns with our API schema definition here too.

```
$ curl -s https://raw.githubusercontent.com/cloudflare/api-schemas/main/openapi.yaml | yq -y '.components.schemas["dns-records_NSRecord"]'

allOf:
  - properties:
      content:
        description: A valid name server host name.
        example: ns1.example.com
      name:
        $ref: '#/components/schemas/dns-records_name'
      type:
        type: string
        description: Record type.
        enum:
          - NS
        example: NS
  - $ref: '#/components/schemas/dns-records_base'
title: NS Record
required:
  - type
  - name
  - content
```

the `type` value being hardcoded to "URI" is also definitely incorrect.

**Comment by jacobbednarz** [comment]
this has been resolved in 3.0.0-beta.10
